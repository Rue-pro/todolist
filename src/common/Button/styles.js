import styled, { css } from 'styled-components'
import { colors } from '../constants'

export const ButtonStyled = styled.div`
  font-weight: 500;
  padding: 12px 18px;
  min-width: 130px;
  box-sizing: border-box;
  border-radius: 23px;
  ${(props) => {
    console.log(props)
    const { primary } = props
    return css`
      color: ${primary ? '#ffffff' : 'yellow'};
      background-color: ${primary ? `rgb(${colors.primary})` : 'yellow'};
    `
  }}
`
