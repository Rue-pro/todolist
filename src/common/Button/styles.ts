import styled, { css } from 'styled-components'
import { colors } from '../constants'
import { ButtonTypeEnum } from './Button'

interface IButtonStyledProps {
  btnType: ButtonTypeEnum
  disabled?: boolean
}

export const ButtonStyled = styled.button<IButtonStyledProps>`
  box-sizing: border-box;
  width: 100%;
  min-width: 130px;
  padding: 12px 18px;
  border: none;
  border-radius: 23px;
  outline: none;
  cursor: pointer;

  & > span {
    color: #ffffff;
  }

  ${(props) => {
    const { btnType, disabled } = props
    if (!disabled) {
      return css`
        background: linear-gradient(45deg, ${colors.btn_primary_focus}, ${colors.btn_primary});
        background-color: ${colors.btn_primary};
        &.hover,
        &:hover {
          background: linear-gradient(45deg, ${colors.btn_primary_focus}, ${colors.btn_primary_hover});
          background-color: ${(btnType === ButtonTypeEnum.primary && `${colors.btn_primary_hover}`) || 'black'};
        }

        &.focus,
        &:focus {
          background: linear-gradient(45deg, ${colors.btn_primary_active}, ${colors.btn_primary_focus});
          background-color: ${(btnType === ButtonTypeEnum.primary && `${colors.btn_primary_focus}`) || 'black'};
        }

        &.active,
        &:active {
          background: linear-gradient(45deg, ${colors.btn_primary_hover}, ${colors.btn_primary_active});
          background-color: ${(btnType === ButtonTypeEnum.primary && `${colors.btn_primary_active}`) || 'black'};
        }
      `
    }
    return css`
      background-color: ${(btnType === ButtonTypeEnum.primary && `${colors.btn_primary_disable}`) || 'black'};
    `
  }}
`
