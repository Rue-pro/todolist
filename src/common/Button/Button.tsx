import React from 'react'
import Text from '../Text/Text'
import { ButtonStyled } from './styles'

export enum ButtonTypeEnum {
  primary
}

export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode
  type?: ButtonTypeEnum
  disabled?: boolean
}

const Button = ({ children, type, disabled, ...rest }: IButtonProps): JSX.Element => {
  return (
    <ButtonStyled disabled={disabled} btnType={type || ButtonTypeEnum.primary} {...rest}>
      <Text>{children}</Text>
    </ButtonStyled>
  )
}

export default Button
