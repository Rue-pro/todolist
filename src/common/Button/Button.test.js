import React from 'react'
import Button from './Button'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { ButtonTypeEnum } from './Button'

describe('Button', () => {
  it('should render correctly with type=primary and disabled=true', () => {
    const output = shallow(
      <Button type={ButtonTypeEnum.primary} disabled={true}>
        primary button
      </Button>
    )
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with type=primary and no disabled', () => {
    const output = shallow(<Button type={ButtonTypeEnum.primary}>primary button</Button>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with wrong props type', () => {
    const output = shallow(<Button type="primary">wrong type button</Button>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with no props children', () => {
    const output = shallow(<Button type={ButtonTypeEnum.primary}></Button>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with no props', () => {
    const output = shallow(<Button />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
