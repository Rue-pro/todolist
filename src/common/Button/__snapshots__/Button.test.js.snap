// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly with no props 1`] = `
<styled.button
  btnType={0}
>
  <Text />
</styled.button>
`;

exports[`Button should render correctly with no props children 1`] = `
<styled.button
  btnType={0}
>
  <Text />
</styled.button>
`;

exports[`Button should render correctly with type=primary and disabled=true 1`] = `
<styled.button
  btnType={0}
  disabled={true}
>
  <Text>
    primary button
  </Text>
</styled.button>
`;

exports[`Button should render correctly with type=primary and no disabled 1`] = `
<styled.button
  btnType={0}
>
  <Text>
    primary button
  </Text>
</styled.button>
`;

exports[`Button should render correctly with wrong props type 1`] = `
<styled.button
  btnType="primary"
>
  <Text>
    wrong type button
  </Text>
</styled.button>
`;
